worlds/tloz/Rules.py

```diff
@@ -0,0 +1,145 @@
+from typing import TYPE_CHECKING
+
+from ..generic.Rules import add_rule
+from .Locations import food_locations, shop_locations
+from .ItemPool import dangerous_weapon_locations
+
+if TYPE_CHECKING:
+    from . import TLoZWorld
+
+def set_rules(tloz_world: "TLoZWorld"):
+    player = tloz_world.player
+    world = tloz_world.multiworld
+
+    # Boss events for a nicer spoiler log play through
+    for level in range(1, 9):
+        boss = world.get_location(f"Level {level} Boss", player)
+        boss_event = world.get_location(f"Level {level} Boss Status", player)
+        status = tloz_world.create_event(f"Boss {level} Defeated")
+        boss_event.place_locked_item(status)
+        add_rule(boss_event, lambda state: state.can_reach(boss, "Location", player))
```

Comment:
as below, this is creating the same rule for all locations. Something like this should fix it:
`add_rule(boss_event, lambda state, b=boss: state.can_reach(b, "Location", player))`